pkg/app/pipedv1/plugin/kubernetes/deployment/determine_test.go

```diff
@@ -1064,3 +1065,148 @@ data:
 		})
 	}
 }
+
+func TestCheckImageChange(t *testing.T) {
+	tests := []struct {
+		name   string
+		old    string
+		new    string
+		want   string
+		wantOk bool
+	}{
+		{
+			name: "image updated",
+			old: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.3
+`,
+			new: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.4
+`,
+			want:   "Sync progressively because of updating image nginx from 1.19.3 to 1.19.4",
+			wantOk: true,
+		},
+		{
+			name: "image name changed",
+			old: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.3
+`,
+			new: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: redis
+        image: redis:6.0.9
+`,
+			want:   "Sync progressively because of updating image nginx:1.19.3 to redis:6.0.9",
+			wantOk: true,
+		},
+		{
+			name: "no image change",
+			old: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.3
+`,
+			new: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.3
+`,
+			want:   "",
+			wantOk: false,
+		},
+		{
+			name: "multiple image updates",
+			old: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.3
+      - name: redis
+        image: redis:6.0.9
+`,
+			new: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.4
+      - name: redis
+        image: redis:6.0.10
+`,
+			want:   "Sync progressively because of updating image nginx from 1.19.3 to 1.19.4, image redis from 6.0.9 to 6.0.10",
+			wantOk: true,
+		},
```

Comment:
Could you add a test case to show that the order of the image matters to this checkImageChange function 👀 
```golang
{
			name: "change the order cause multi-image update",
			old: `
apiVersion: apps/v1
kind: Deployment
metadata:
  name: app-deployment
spec:
  template:
    spec:
      containers:
      - name: nginx
        image: nginx:1.19.3
      - name: redis
        image: redis:6.0.9
`,
			new: `
apiVersion: apps/v1
kind: Deployment
metadata:
  name: app-deployment
spec:
  template:
    spec:
      containers:
      - name: redis
        image: redis:6.0.9
      - name: nginx
        image: nginx:1.19.3
`,
			want:   "Sync progressively because of updating image nginx:1.19.3 to redis:6.0.9, image redis:6.0.9 to nginx:1.19.3",
			wantOk: true,
		},
```

Reply:
Thanks. I added it on this commit.
https://github.com/pipe-cd/pipecd/pull/5333/commits/1112cd96fa16c0f7d567304dcd694487edb47443